### PR TITLE
Transformers edits and updates (WIP)

### PR DESCRIPTION
**Pull request by saidbleik**
### Related Issues

https://github.com/microsoft/nlp/pull/434

### Checklist:

- [ ] My code follows the code style of this project, as detailed in our [contribution guidelines](../CONTRIBUTING.md).
- [ ] I have added tests.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: utils_nlp/models/transformers/common.py [hunk 1]
```diff
@@ -34,9 +34,9 @@
 logger = logging.getLogger(__name__)
 
 
-def get_device(device, num_gpus, local_rank):
+def get_device(num_gpus=None, local_rank=-1):
```
Comment (miguelgfierro) [inline utils_nlp/models/transformers/common.py]: after this change, in `fine_tune` and `predict` do we still need the parameter device?

File: utils_nlp/models/transformers/common.py [hunk 2]
```diff
@@ -58,35 +57,31 @@ def __init__(
         cache_dir=".",
         load_model_from_dir=None,
     ):
-        self.model_name = model_name
+
+        if model_name not in self.list_supported_models():
+            raise ValueError(
+                "Model name {0} is not supported by {1}. "
+                "Call '{2}.list_supported_models()' to get all supported model "
+                "names.".format(value, self.__class__.__name__, self.__class__.__name__)
```
Comment (hlums) [inline utils_nlp/models/transformers/common.py]: This was a mistake I made. We only need to keep self.\_\_class\_\_.\_\_name\_\_ and change {2} to {1}.

File: utils_nlp/models/transformers/common.py [hunk 3]
```diff
@@ -34,9 +34,9 @@
 logger = logging.getLogger(__name__)
 
 
-def get_device(device, num_gpus, local_rank):
+def get_device(num_gpus=None, local_rank=-1):
     if local_rank == -1:
-        device = torch.device("cuda" if torch.cuda.is_available() and device == "cuda" else "cpu")
+        device = torch.device("cuda" if torch.cuda.is_available() and num_gpus > 0 else "cpu")
         num_gpus = (
```
[the remark this answers is not in the document]
Comment (miguelgfierro) [reply utils_nlp/models/transformers/common.py]: mmm I realized that we have another `get_device` in https://github.com/microsoft/nlp/blob/transformers2/utils_nlp/common/pytorch_utils.py I'll make the changes to have this in the same place

[the remark this answers is not in the document]
Comment (miguelgfierro) [reply utils_nlp/models/transformers/common.py]: PR here https://github.com/microsoft/nlp/pull/452

File: utils_nlp/models/transformers/common.py (hunk 1 above)
Comment (miguelgfierro) [reply utils_nlp/models/transformers/common.py]: > after this change, in fine_tune and predict do we still need the parameter device?

I think the device is computed in the child classes and then is passed into fine_tune and predict. We can leave it like this or maybe by DRY compute it in the base class, but it is not a big deal

Comment (saidbleik) [reply utils_nlp/models/transformers/common.py]: I'll leave it like this. It's not exposed to the user anyway.

**Pull request merged**